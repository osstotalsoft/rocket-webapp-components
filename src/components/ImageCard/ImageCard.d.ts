export interface ImageCardProps {

  /**
   * Override or extend the styles applied to the component
   */
  className?: string;

  /**
   * The content of the card
   */
  children?: React.ReactNode;

  /**
   * The image to be displayed
   */
  image?: string | React.ReactNode;

  /**
   * If the image cannot be displayed for some reason, this text will be shown
   */
  alternateText?: string;

  /**
   * The color of the card
   */
  color?: "primary" |
          "info" |
          "success" |
          "warning" |
          "danger" |
          "rose"

  /**
   * @default 's'
   * The size of the image
   */
  imageSize?: 's' | 'm' | 'l'

  /**
   * Demos:
   * 
   * - https://bit.dev/totalsoft_oss/react-mui/image-card
   * 
   *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/ImageCard/ImageCard.js
   */

}

export default function ImageCard(props: ImageCardProps): JSX.Element;