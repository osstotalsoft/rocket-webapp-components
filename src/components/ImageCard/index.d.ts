export { default } from './ImageCard';
export * from './ImageCard';