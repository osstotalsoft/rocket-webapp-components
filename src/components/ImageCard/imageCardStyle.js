import styles from '../../assets/jss/styles.js';
import {theme as defaultTheme} from '../themes/defaultTheme';

const imageCardStyle = theme => {
  const {
    whiteColor,
    roseColor,
    successColor,
    dangerColor,
    warningColor,
    infoColor,
    blackColor,
    hexToRgb
  } = styles(theme);

  return ({
    image: {
      borderRadius: "50% !important",
      margin: '5px 0px',
      boxShadow:
        "0 5px 15px -8px rgba(" +
        hexToRgb(blackColor) +
        ", 0.24), 0 8px 10px -5px rgba(" +
        hexToRgb(blackColor) +
        ", 0.2)"
    },
    s: {
      height: '80px',
      width: '80px'
    },
    m: {
      height: '163px',
      width: '163px'
    },
    l: {
      height: '280px',
      width: '280px'
    },
    primary: {
      background: theme.palette?.primary?.main || defaultTheme.palette?.primary?.main,
      color: whiteColor
    },
    info: {
      background:
        "linear-gradient(60deg," + infoColor[1] + "," + infoColor[2] + ")",
      "& h1 small": {
        color: "rgba(" + hexToRgb(whiteColor) + ", 0.8)"
      },
      color: whiteColor
    },
    success: {
      background:
        "linear-gradient(60deg," + successColor[1] + "," + successColor[2] + ")",
      "& h1 small": {
        color: "rgba(" + hexToRgb(whiteColor) + ", 0.8)"
      },
      color: whiteColor
    },
    warning: {
      background:
        "linear-gradient(60deg," + warningColor[1] + "," + warningColor[2] + ")",
      "& h1 small": {
        color: "rgba(" + hexToRgb(whiteColor) + ", 0.8)"
      },
      color: whiteColor
    },
    danger: {
      background:
        "linear-gradient(60deg," + dangerColor[1] + "," + dangerColor[2] + ")",
      "& h1 small": {
        color: "rgba(" + hexToRgb(whiteColor) + ", 0.8)"
      },
      color: whiteColor
    },
    rose: {
      background:
        "linear-gradient(60deg," + roseColor[1] + "," + roseColor[2] + ")",
      "& h1 small": {
        color: "rgba(" + hexToRgb(whiteColor) + ", 0.8)"
      },
      color: whiteColor
    }
  })
};

export default imageCardStyle;