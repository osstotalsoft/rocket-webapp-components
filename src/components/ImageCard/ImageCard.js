import React from "react";
import PropTypes from "prop-types";
import { Grid, makeStyles } from "@material-ui/core";
import Card from "../Card";
import imageCardStyle from "./imageCardStyle";

const useStyles = makeStyles(imageCardStyle);

function ImageCard({ children, image, imageSize, alternateText, ...rest }) {
  const classes = useStyles();

  return (
    <Card {...rest}>
      <Grid container justifyContent="center">
        <img
          className={classes.image + " " + classes[imageSize]}
          src={image}
          alt={alternateText}
        />
      </Grid>
      {children}
    </Card>
  );
}

ImageCard.defaultProps = {
  imageSize: "s"
};

ImageCard.propTypes = {
  /**
  * Override or extend the styles applied to the component
  */
  className: PropTypes.string,
  /**
  * The content of the card
  */
  children: PropTypes.node,
  /**
  * The image to be displayed
  */
  image: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  /**
  * If the image cannot be displayed for some reason, this text will be shown
  */  alternateText: PropTypes.string,
  /**
  * The color of the card
  */
  color: PropTypes.oneOf([
    "primary",
    "info",
    "success",
    "warning",
    "danger",
    "rose"
  ]),
  /**
  * The size of the image
  */
  imageSize: PropTypes.oneOf(["s", "m", "l"])
};

export default ImageCard;
