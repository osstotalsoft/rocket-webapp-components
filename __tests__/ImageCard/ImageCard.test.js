import React from "react";
import ImageCard from "../../components/ImageCard/ImageCard";
import AssignmentLate from "@material-ui/icons/AssignmentLate";
import { mount } from 'enzyme';

describe("Render ImageCard", () => {
  it("imageSize should be equal to 's'", () => {
    const wrapper = mount(
      <ImageCard
        image={AssignmentLate}
        alternateText="Image card"
      />
    );

    expect(wrapper.props().imageSize).toBe("s")
  });

  it("alternateText prop should be present in the html attribute alt", () => {
    const wrapper = mount(
      <ImageCard
        image={AssignmentLate}
        alternateText="Image card"
      />
    );

    expect(wrapper.find('img').props().alt).toBe("Image card")
  });
});
