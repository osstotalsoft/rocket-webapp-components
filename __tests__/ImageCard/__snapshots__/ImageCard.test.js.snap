// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ImageCard snapshot 1`] = `
<div
  className="makeStyles-card-11"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
  >
    <img
      alt="Image card"
      className="makeStyles-image-1 makeStyles-m-3"
      src={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        }
      }
    />
  </div>
</div>
`;
